client/src/client.rs

```diff
@@ -1151,7 +1157,12 @@ fn log_response(cmd: &str, resp: &Result<jsonrpc::Response>) {
                         debug!(target: "bitcoincore_rpc", "JSON-RPC error for {}: {:?}", cmd, e);
                     }
                 } else if log_enabled!(Trace) {
-                    let result = resp.result.as_ref().unwrap_or(&serde_json::Value::Null);
+                    // we can't use to_raw_value here due to compat with Rust 1.29
+                    let def = serde_json::value::RawValue::from_string(
```

Comment:
This should happen in a closure given to `unwrap_or_else` so it is only run when needed.

Reply:
Ha right. I'm used to Haskell's pervasive laziness. Will fix.

Reply:
If I don't allocate it here, then I have to clone the response value which is strictly worse. This value is 4 bytes.

Reply:
Alternatively I could use a Cow, but that is 16 bytes...

Reply:
Because the response value is borrowed, the alternative (Null here) must also be borrowed. If I allocate it inside the unwrap_or_else, the value will be dropped while it is borrowed. Here are my options:

1. Unify borrowed and owned values through a Cow (costing 16bytes in both optimistic and pessimistic cases)
2. Leave it as is (costing 4 bytes in both optimistic and pessimistic cases)
3. Keep the Null as a lazy_static (amortizing the cost of the Null to zero, also pulls in lazy_static dependency)
4. Clone the response value (can be massive in the cases of large responses like `getrawblock`)

Reply:
Idk if cow is really worth it then, I vote for leave it as is and try to re-engineer the API later on, it's kind of a mess anyway.